dap-mode.el

```diff
@@ -1222,6 +1224,7 @@ DEBUG-SESSION is the active debug session."
         (run-hooks 'dap-breakpoints-changed-hook)))))
 
 (defun dap--breakpoint-filter-enabled (filter type default)
+  (defvar dap-exception-breakpoints) ;; NOTE: always called with dap-ui loaded
```

Comment:
This should be moved to top level since we need this only for byte-compile. Just one time evaluation at top level is enough. You can see other file in this project is doing the same.

Reply:
I was waiting for that comment :). That defvar has no overhead. Its not even in the byte-code. Evaluate the following in scratch and you'll see what I mean:
```emacs-lisp
(disassemble (byte-compile '(lambda () (defvar foo) 500)))
```
-> constant 500; return
Defvar without a default-value is just a byte-compiler hint that it is a global variable and will be imported and let-bindings should always override. (Similar to declare-function).

Reply:
Well, I think that the problem here is the interpreted scope.
With `defvar` inside `defun` it will be understood as only available/effective inside that function.
If there's another one which also use this variable, it will need to `defvar` that variable too. No much different from `boundp` check.

Having all forward declaration in top level will prevent that, at least makes it clear that the variable is available in whole file scope already.

Reply:
Well, one can always evil-search-forward? Having such declare-functions close to requires makes it clear what is actually imported, but what do you think?

Reply:
I think we can move the `defvar` to just a few line up, right before `defun`. In that way, it's still close to where it's first used + we will have it as top level declaration. And we don't have to write `defvar` or `declare-function` every time it being used in another function. To keep it consistent with the convention already used in this project.

Reply:
Ahh, if there is a convention I'll update it soon. I am however currently busy with the refactor to move dap-ui--show-buffer to display-buffer.